sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -572,6 +603,18 @@ case class Range(
 
   require(step != 0, s"step ($step) cannot be 0")
 
+  def this(start: Long, end: Long, step: Long, numSlices: Int) =
```

Comment:
A function-like `Expression` should have constructors that accept `Expression` inputs. There are other functions that only take constant as inputs, but the constructor still takes `Expression`. See `HyperLogLogPlusPlus`.